main.tf

```diff
@@ -256,6 +256,14 @@ resource "aws_kms_alias" "this" {
   target_key_id = var.create_external ? aws_kms_external_key.this[0].id : aws_kms_key.this[0].key_id
 }
 
+resource "aws_kms_alias" "computed" {
+  for_each = { for k, v in var.computed_aliases : k => v if var.create }
```

Comment:
Can't this one be merged with `this` from above? Something like this but with the proper list-to-map conversion of `var.aliases`:

```
  for_each = { for k, v in merge(var.aliases, var.computed_aliases) : k => v if var.create }
```

IIRC, as soon as `k` is not unknown but computed, this will work fine.